plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfig.java

```diff
@@ -42,6 +47,39 @@ public S3SecurityMappingConfig setConfigFile(File configFile)
         return this;
     }
 
+    public Optional<String> getConfigUri()
+    {
+        return Optional.ofNullable(configUri);
+    }
+
+    @Config("hive.s3.security-mapping.config-uri")
+    @ConfigDescription("URI returning the security mappings as JSON")
+    public S3SecurityMappingConfig setConfigUri(String configUri)
+    {
+        this.configUri = configUri;
+        return this;
+    }
+
+    @AssertTrue(message = "Cannot set both hive.s3.security-mapping.config-file and hive.s3.security-mapping.config-uri")
+    public boolean atMostOneProvider()
+    {
+        return getConfigFile().isEmpty() || getConfigUri().isEmpty();
+    }
+
+    @NotNull
+    public String getJSONPointer()
```

Comment:
Our code style is to name things in camel case where only the first letter is capitalized, including acronyms, so write this as `getJsonPointer`